### PR TITLE
fix redraw issue when logging in to or out of a domain

### PR DESCRIPTION
**Pull request by ZappoMan**

### CODE REVIEW

**Comment by ZappoMan** [comment]
Testing Notes:
- Run interface, go to a domain that's open/doesn't require login
- If you start logged in - wait till you see domain content - 
- then log out - you should see the domain content disappear (that's correct) and then reappear (that's the fix)
- log back in again - you should see the domain content disappear (that's correct) and then reappear (that's the fix)

In the production builds, you would not see content re-appear unless you moved or rotated.

This fixes this bug: https://app.asana.com/0/26320649018758/105153848426941

**Comment by ZappoMan** [comment]
Discussion of the bug/fix:

The bug is caused by a couple of factors. Apparently when logging out/logging in - the nodeID is apparently not changing, and so from the entity servers perspective, the client is the same client... and so it honors it's last know query, and won't resend content unless the content has changed or the query  has changed significantly.

The fix is to fake a dramatic change in the query in this case.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 47aed9c8bd8996de2ebd514c290194778454e65c available at: https://gist.github.com/2ab2925d0b0e3d56fe528eb043a25fc2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7604.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7604.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -4173,10 +4182,19 @@ void Application::nodeActivated(SharedNodePointer node) const {
             }
         }
     }
+
+    // If we get a new EntityServer activated, do a "forceRedraw" query. This will send a degenerate
+    // query so that if the server doesn't
```
Comment (jherico) [inline interface/src/Application.cpp]: incomplete comment

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5f4d6ca1d31efe47697c82b4082831817b305d1d available at: https://gist.github.com/2ab2925d0b0e3d56fe528eb043a25fc2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7604.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7604.dmg

**Comment by Triplelexx** [comment]
worked consistently for me.

**Comment by huffman** [comment]
Already approved, but I'm curious why we aren't clearing the node state on the entity server when a node disconnects? Is there any value in maintaining it?

**Comment by ZappoMan** [comment]
If the entity server saw a disconnect it would clear the state. Since the node uuid for the client would change. 

But when I checked the entity server doesn't see this as a disconnect for some reason. I think this node killed node added is only happening on the client. 

Let me double check.

**Comment by ZappoMan** [comment]
Starting with client "logged out"....
```
[04/07 08:26:46] [DEBUG] [4232] [entity-server] Added "Agent" (I) {79c9bb06-bf03-4239-a759-84164548b6f0} 174.21.92.4:61986 / 192.168.0.11:61986 
[04/07 08:26:46] [DEBUG] [4232] [entity-server] Entity server added node: "Agent" (I) {79c9bb06-bf03-4239-a759-84164548b6f0} 174.21.92.4:61986 / 192.168.0.11:61986 
[04/07 08:26:46] [DEBUG] [4232] [entity-server] Activating local socket for network peer with ID "79c9bb06-bf03-4239-a759-84164548b6f0"
[04/07 08:26:46] [DEBUG] [4232] [entity-server] Entity server [ EntityServer(0xc2c1e8a660) ]: client connected - starting sending thread [ OctreeSendThread(0xc2c3c48a60, name = "Octree Send Thread (79c9bb06-bf03-4239-a759-84164548b6f0)") ]
[04/07 08:26:46] [DEBUG] [7636] [avatar-mixer] Added "Agent" (I) {79c9bb06-bf03-4239-a759-84164548b6f0} 174.21.92.4:61986 / 192.168.0.11:61986 
[04/07 08:26:46] [DEBUG] [7636] [avatar-mixer] Activating local socket for network peer with ID "79c9bb06-bf03-4239-a759-84164548b6f0"
[04/07 08:26:46] [DEBUG] [7636] [avatar-mixer] Changing skeleton model for avatar to "file:///C:/Development/HiFi/hifi/build/interface/Release/resources/meshes/defaultAvatar_full.fst"
[04/07 08:26:46] [DEBUG] [2160] [messages-mixer] Added "Agent" (I) {79c9bb06-bf03-4239-a759-84164548b6f0} 174.21.92.4:61986 / 192.168.0.11:61986 
[04/07 08:26:46] [DEBUG] [2160] [messages-mixer] Activating local socket for network peer with ID "79c9bb06-bf03-4239-a759-84164548b6f0"
[04/07 08:26:46] [DEBUG] [1340] [asset-server] Added "Agent" (I) {79c9bb06-bf03-4239-a759-84164548b6f0} 174.21.92.4:61986 / 192.168.0.11:61986 
[04/07 08:26:46] [DEBUG] [1340] [asset-server] Activating local socket for network peer with ID "79c9bb06-bf03-4239-a759-84164548b6f0"
[04/07 08:26:46] [DEBUG] [15916] [audio-mixer] Added "Agent" (I) {79c9bb06-bf03-4239-a759-84164548b6f0} 174.21.92.4:61986 / 192.168.0.11:61986 
[04/07 08:26:46] [DEBUG] [15916] [audio-mixer] Activating local socket for network peer with ID "79c9bb06-bf03-4239-a759-84164548b6f0"
```
Then when the client logs in...
```
[04/07 08:27:37] [DEBUG] [15916] [audio-mixer] unreasonable sequence number: 0 previous: 4721
[04/07 08:27:37] [DEBUG] [15916] [audio-mixer] re-synced with sequence number sender
[04/07 08:27:38] [SUPPRESS] [15916] [audio-mixer] 7 repeated log entries matching "unreasonable sequence number: \d+ previous: \d+" - Last entry: "unreasonable sequence number: 7 previous: 4721"
```
Then after being logged in... if you check the "nodes" page on the console/domain server, you will see that the UUID for the client is still "79c9bb06-bf03-4239-a759-84164548b6f0"

**Comment by ZappoMan** [comment]
I also notice that even after login - the domain server doesn't report the user as "logged in" in the "nodes" page... so clearly there are other bugs. And maybe if after fixing those bugs, this client side work around for the blank scene may be unneeded. I will file the domain server bug.

**Comment by huffman** [comment]
Agreed, we should get any of those networking issues fixed. I'm also concerned if we aren't being notified of disconnects on both ends, could be a protocol team issue that we'll need to address.

**Pull request merged**